daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/dfa/Runtime.scala

```diff
@@ -147,6 +159,9 @@ trait DFADelimiter extends DFA {
   final val isES = lookingFor == "%ES;"
 
   def unparseValue: String
+
+  def strForDiagnostic = DFADelimiter.strForDiagnostic(lookingFor)
```

Comment:
Control characters like NUL and Ctrl-B, STX/ETX, STT, EOT, ESC, DEL, etc. are also used as delimiters quite a bit. 

ALso, I think you can do this: dfdl:terminator="&#x20; &#x09;" which is a space and a tab character. XML allows it. I am not sure DFDL won't normalize that away and reject the definition, but either way you can have other quasi-whitespace control chars that would be confusing to display.

Reply:
Seems odd to use actual control characters instead of character class equivalents, but I guess it's possible?  Should we then change `lookingFor` so that is the remapped version, an remap it a compile time? lookingFor seems to be the diagnostic, or is it use for other things?

Ca we also make this a val or pass it in as a constructor paramter?  We probably shouldn't remap this everytime it ends up in diagnostics, it's normal behavior to find an out of scope delimiter and need to create a PE.

Reply:
This should be resolved. I changed this to lazy val so there is no allocation per use. I did not chase down all the lookingFor to see if they are all just diagnostic display, as there were a couple cases where it wasn't obvious at first glance.